### ISSUE TITLE
FileSystemWatcher_Changed_PreSeededNestedStructure failed in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#2714]
One of the tests https://github.com/dotnet/corefx/pull/2682 was aiming to fix still just failed in CI.

http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug_prtest/2919/console
```
ChangedTests.FileSystemWatcher_Changed_PreSeededNestedStructure(includeSubdirectories: True) [FAIL]
        Didn't observe a second file change event within 500ms
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.IO.FileSystem.Watcher\tests\Utility\Utility.cs(103,0): at Utility.ExpectEvent(WaitHandle eventOccured, String eventName, Int32 timeout)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.Changed.cs(188,0): at ChangedTests.FileSystemWatcher_Changed_PreSeededNestedStructure(Boolean includeSubdirectories)
```

**Comment by stephentoub** [comment]
Along with other FSW tests in a similar run:
```
NotifyFilterTests.FileSystemWatcher_NotifyFilter_Attributes [FAIL]
        Didn't observe a changed event within 500ms
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\Utility\Utility.cs(103,0): at Utility.ExpectEvent(WaitHandle eventOccured, String eventName, Int32 timeout)
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.NotifyFilter.cs(30,0): at NotifyFilterTests.FileSystemWatcher_NotifyFilter_Attributes()
     NotifyFilterTests.FileSystemWatcher_NotifyFilter_CreationTime [FAIL]
        Didn't observe a changed event within 500ms
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\Utility\Utility.cs(103,0): at Utility.ExpectEvent(WaitHandle eventOccured, String eventName, Int32 timeout)
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.NotifyFilter.cs(50,0): at NotifyFilterTests.FileSystemWatcher_NotifyFilter_CreationTime()
     ErrorTests.FileSystemWatcher_Error_File [FAIL]
        Didn't observe a error event within 500ms
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\Utility\Utility.cs(103,0): at Utility.ExpectEvent(WaitHandle eventOccured, String eventName, Int32 timeout)
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.Error.cs(74,0): at ErrorTests.FileSystemWatcher_Error_File()
     DeletedTests.FileSystemWatcher_Deleted_File [FAIL]
        Didn't observe a deleted event within 500ms
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\Utility\Utility.cs(103,0): at Utility.ExpectEvent(WaitHandle eventOccured, String eventName, Int32 timeout)
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.Deleted.cs(27,0): at DeletedTests.FileSystemWatcher_Deleted_File()
     CreatedTests.FileSystemWatcher_Created_File [FAIL]
        Didn't observe a created event within 500ms
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\Utility\Utility.cs(103,0): at Utility.ExpectEvent(WaitHandle eventOccured, String eventName, Int32 timeout)
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.Created.cs(26,0): at CreatedTests.FileSystemWatcher_Created_File()
     ChangedTests.FileSystemWatcher_Changed_PreSeededNestedStructure(includeSubdirectories: True) [FAIL]
        Didn't observe a second file change event within 500ms
        Expected: True
        Actual:   False
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\Utility\Utility.cs(103,0): at Utility.ExpectEvent(WaitHandle eventOccured, String eventName, Int32 timeout)
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.Changed.cs(188,0): at ChangedTests.FileSystemWatcher_Changed_PreSeededNestedStructure(Boolean includeSubdirectories)
```
If this looks to be more than just a fleeting issue, I'll back out 2682 until we sort it out.

**Comment by stephentoub** [comment]
Fixed by #2716
